ludwig/features/date_feature.py

```diff
@@ -114,15 +114,14 @@ def add_feature_data(
 
 @register_input_feature(DATE)
 class DateInputFeature(DateFeatureMixin, InputFeature):
-    encoder = "embed"
-
-    def __init__(self, feature, encoder_obj=None):
-        super().__init__(feature)
-        self.overwrite_defaults(feature)
+    def __init__(self, input_feature_config: DateInputFeatureConfig, encoder_obj=None, **kwargs):
+        input_feature_config = self.load_config(input_feature_config)
```

Comment:
Is it necessary to call `load_config` here? If the `input_feature_config` is always a `DateInputFeatureConfig`, then it should be a no-op, right?

Reply:
So I believe @dantreiman implemented this section so I will defer to him for the architecture choice here. What I can say is that we should be able to remove this once the config object PR is landed since the full config object will be initialized earlier in the pipeline so we won't have to load the individual configs here in the feature initializers.

Reply:
Woops.  I meant to declare the type of `input_feature_config` as `Union[dict, DateInputFeatureConfig]`.  I intended to keep that initialization path working until we finally finish schema-fication because many of our tests construct input features from a config dict.